files/usr/lib/jeos-firstboot

```diff
@@ -131,62 +131,89 @@ findlocales()
-[ -n "$locale_lang" ] && default="$locale_lang"
+if [ -z "$LOCALE_PRESET" ]; then
+    default="en_US"
+    [ -f /etc/locale.conf ] && locale_lang="$(awk -F= '$1 == "LANG" { split($2,fs,"."); print fs[1]; exit }' /etc/locale.conf)"
+    [ -n "$locale_lang" ] && default="$locale_lang"
+
+    if findlocales && d --default-item "$default" --menu  $"Select System Locale" 0 0 $dh_menu "${list[@]}"; then
+        if [ -n "$result" ]; then
+            locale="${result}.UTF-8"
+            systemd_firstboot_args+=("--locale=$locale")
+            LANG="$locale"
+            export LANG
+
+                # Run langset to get consolefont and keymap set up
+                run langset.sh $locale || warn $"Setting the locale failed"
+        fi
+    else
+        d --msgbox $"Error setting locale" 0 0
+    fi
+else
+    locale="$LOCALE_PRESET"
+    systemd_firstboot_args+=("--locale=$locale")
+fi
 
-if findlocales && d --default-item "$default" --menu  $"Select System Locale" 0 0 $dh_menu "${list[@]}"; then
-	if [ -n "$result" ]; then
-	    locale="${result}.UTF-8"
-	    systemd_firstboot_args+=("--locale=$locale")
-	    LANG="$locale"
-	    export LANG
+if [ -z "$KEYTABLE_PRESET" ]; then
+    default="us"
+    [ -f /etc/vconsole.conf ] && vconsole_keymap="$(awk -F= '$1 == "KEYMAP" { split($2,fs,"."); print fs[1]; exit }' /etc/vconsole.conf)"
+    [ -n "$vconsole_keymap" ] && default="$vconsole_keymap"
 
-            # Run langset to get consolefont and keymap set up
-            run langset.sh $locale || warn $"Setting the locale failed"
-	fi
+    if findkeymaps \
+        && d --default-item "$default" --menu  $"Select Keyboard Layout" 0 0 $dh_menu "${list[@]}"; then
+        if [ -n "$result" ]; then
+            keytable="$result"
+
+                # Activate the selected keyboard layout
+                run langset.sh "$locale" "$keytable" || warn $"Setting the keyboard layout failed"
+        fi
+    else
+        d --msgbox $"Error setting keyboard" 0 0
+    fi
 else
-	d --msgbox $"Error setting locale" 0 0
+    keytable="$result"
 fi
 
-if [ -e "$EULA_FILE" -a ! -e "${EULA_FILE%/*}/no-acceptance-needed" ]; then
-    [ -n "${locale}" ] && language="${locale%%_*}" || language="en"
+[ -n "${locale}" ] && language="${locale%%_*}" || language="en"
+force_english_license=0
 
+if [[ "$(tty)" = /dev/tty* ]]; then
     # Those languages can't be displayed in the console
-    declare -A use_english_license
-    use_english_license["ja"]=1
-    use_english_license["zh"]=1
-    use_english_license["ko"]=1
-
-    if [ -n "$locale" -a -z "${use_english_license[${language}]+_}" ]; then
-	for i in "${EULA_FILE%.txt}.$locale.txt" \
-	         "${EULA_FILE%.txt}.${locale%%.UTF-8}.txt" \
-	         "${EULA_FILE%.txt}.${language}.txt"; do
-	    if [ -e "$i" ]; then
-		EULA_FILE="$i"
-		break
-	    fi
-	done
+    declare -A start_kmscon
+    start_kmscon["cs"]=1
```

Comment:
Why `cs`?

Reply:
license.cs.txt from openSUSE at least contains characters the console font does not contain.